modules/memfault-firmware-sdk/memfault_location_metrics.c

```diff
@@ -0,0 +1,209 @@
+#include <modem/location.h>
+
+#include "memfault/metrics/metrics.h"
+#include "memfault_location_metrics.h"
+
+LOG_MODULE_DECLARE(memfault_ncs_metrics, CONFIG_MEMFAULT_NCS_LOG_LEVEL);
+
+static bool s_session_in_progress;
+
+/**
+ * @brief Convert location method to string
+ *
+ * @param method method to convert
+ * @return const char* method string
+ */
+static const char *prv_location_method_to_string(enum location_method method)
+{
+	switch (method) {
+	case LOCATION_METHOD_CELLULAR:
+		return "Cellular";
+	case LOCATION_METHOD_GNSS:
+		return "GNSS";
+	case LOCATION_METHOD_WIFI:
+		return "WiFi";
+	default:
+		return "Unknown";
+	}
+}
+
+/**
+ * @brief Record location method results
+ *
+ * @param method location method
+ * @param id location event id
+ * @param details pointer to location data details
+ * @param accuracy_m pointer to location accuracy in meters
+ */
+static void prv_location_method_results_record(enum location_method method,
+					       enum location_event_id id,
+					       const struct location_data_details *details,
+					       const float *accuracy_m)
+{
+	switch (method) {
+#if defined(CONFIG_LOCATION_METHOD_GNSS)
+	case LOCATION_METHOD_GNSS:
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_gnss_method_time_ms,
+					     details->elapsed_time_method);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_gnss_method_result, id);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_gnss_on_time_ms,
+					     details->gnss.elapsed_time_gnss);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_gnss_satellites_tracked_count,
+					     details->gnss.satellites_tracked);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_gnss_satellites_used_count,
+					     details->gnss.satellites_used);
+
+		/* Only record TTF and accuracy when a fix is acquired */
+		if (id == LOCATION_EVT_LOCATION) {
+			MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_gnss_time_to_fix_ms,
+						     details->gnss.pvt_data.execution_time);
+
+			if (accuracy_m != NULL) {
+				uint32_t accuracy_cm = (uint32_t)(*accuracy_m * 100.0f);
+
+				MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_accuracy_cm, accuracy_cm);
+			}
+		}
+		break;
+#endif
+#if defined(CONFIG_LOCATION_METHOD_CELLULAR)
+	case LOCATION_METHOD_CELLULAR:
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_lte_method_time_ms,
+					     details->elapsed_time_method);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_lte_method_result, id);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_lte_neighbor_cells_count,
+					     details->cellular.ncells_count);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_lte_gci_cells_count,
+					     details->cellular.gci_cells_count);
+		break;
+#endif
+#if defined(CONFIG_LOCATION_METHOD_WIFI)
+	case LOCATION_METHOD_WIFI:
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_wifi_method_time_ms,
+					     details->elapsed_time_method);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_wifi_method_result, id);
+		MEMFAULT_METRIC_SET_UNSIGNED(ncs_loc_wifi_ap_count,
+					     details->wifi.ap_count);
+		break;
+#endif
+	default:
+		LOG_DBG("Unsupported location method id=%d", method);
```

Comment:
You should handle `LOCATION_METHOD_WIFI_CELLULAR` method too. In that case, details for both cellular and Wi-Fi are available in the structures.

Reply:
Ah good call! I'll add that